src/core/SampleBuffer.cpp

```diff
@@ -1411,25 +1411,35 @@ void SampleBuffer::setReversed( bool _on )
 
 
 
-QString SampleBuffer::tryToMakeRelative( const QString & _file )
+QString SampleBuffer::tryToMakeRelative( const QString & file )
 {
-	if( QFileInfo( _file ).isRelative() == false )
+	if( QFileInfo( file ).isRelative() == false )
 	{
-		QString f = QString( _file ).replace( QDir::separator(), '/' );
-		QString fsd = ConfigManager::inst()->factorySamplesDir();
-		QString usd = ConfigManager::inst()->userSamplesDir();
-		fsd.replace( QDir::separator(), '/' );
-		usd.replace( QDir::separator(), '/' );
-		if( f.startsWith( fsd ) )
+		QString f = QString( file ).replace( QDir::separator(), '/' );
+
+		// First, look in factory samples
+		// Isolate "samples/" from "data:/samples/"
+		QString samplesSuffix = ConfigManager::inst()->factorySamplesDir().mid( ConfigManager::inst()->dataDir().length() );
+
+		// Iterate over all valid "data:/" searchPaths
+		foreach ( const QString& path, QDir::searchPaths( "data" ) )
```

Comment:
Minor, but `foreach` is a candidate for removal in Qt, [quote](http://doc.qt.io/qt-5.8/qtglobal.html#foreach):

> Since Qt 5.7, the use of this macro is discouraged. It will be removed in a future version of Qt. Please use C++11 range-for, possibly with qAsConst(), as needed.

So this would be the preferred way:

```c++
for ( const QString& path : QDir::searchPaths( "data" ) )
```